### PR TITLE
fix: rm dependency on 08-wasm in ibc-go

### PR DESCRIPTION
**Pull request by charleenfei**
## Description

closes: #4779

### Commit Message / Changelog Entry

```text
type: commit message
```

see the [guidelines](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) for commit messages. (view raw markdown for examples)

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/build/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to be used for the changelog entry in the PR description for review.
- [ ] Re-reviewed `Files changed` in the Github PR explorer.
- [ ] Review `Codecov Report` in the comment section below once CI passes.

### CODE REVIEW

**Comment by colin-axner** [comment]
maybe we can remove the code in `testing/endpoint.go` as well?

**Comment by charleenfei** [comment]
> maybe we can remove the code in `testing/endpoint.go` as well? But if it seems like to much work, we can do it in a followup no problem

absolutely, i can do it

**Review by crodriguezvega — APPROVED** [review]
LGTM! I pushed a commit commenting out a bit of code that is also not used anymore.

File: go.mod
Comment (colin-axner) [inline go.mod:1]: the replace statement should be removed?

File: modules/light-clients/08-wasm/types/types_test.go [hunk 1]
```diff
@@ -75,39 +73,39 @@ func setupTestingApp() (ibctesting.TestingApp, map[string]json.RawMessage) {
...
 
-// SetupWasmTendermint sets up 2 chains and stores the tendermint/cometbft light client wasm contract on both.
-func (suite *TypesTestSuite) SetupWasmTendermint() {
-	suite.coordinator = ibctesting.NewCoordinator(suite.T(), 2)
-	suite.chainA = suite.coordinator.GetChain(ibctesting.GetChainID(1))
-	suite.chainA.SetWasm(true)
-	suite.chainB = suite.coordinator.GetChain(ibctesting.GetChainID(2))
-	suite.chainB.SetWasm(true)
+// // SetupWasmTendermint sets up 2 chains and stores the tendermint/cometbft light client wasm contract on both.
+// func (suite *TypesTestSuite) SetupWasmTendermint() {
```
Comment (colin-axner) [inline modules/light-clients/08-wasm/types/types_test.go:77]: Maybe we remove this function now?

File: modules/light-clients/08-wasm/types/update_test.go [hunk 2]
```diff
@@ -458,293 +449,293 @@ func (suite *TypesTestSuite) TestUpdateStateGrandpa() {
...
+// 				})
+// 			}
+
+// 			// perform custom checks
+// 			tc.expResult()
+// 		})
+// 	}
+// }
+
+// func (suite *TypesTestSuite) TestPruneConsensusStateTendermint() {
```
Comment (colin-axner) [inline modules/light-clients/08-wasm/types/update_test.go:655]: feel like this test can be removed, not sure what it was testing in the 08-wasm client?

**Comment by crodriguezvega** [comment]
Looks like [we can also clean up](https://github.com/cosmos/ibc-go/pull/4603/files#r1339866138) the [docker file](https://github.com/cosmos/ibc-go/blob/feat/wasm-clients/Dockerfile#L12-L16).

File: testing/wasm.go [hunk 3]
```diff
@@ -1,64 +1,63 @@
 package ibctesting
```
Comment (colin-axner) [inline testing/wasm.go:1]: I think we can remove this file, the testing pkg should be designed in a way where light clients don't need to put their testing structures here

**Review by colin-axner — APPROVED** [review]
ACK, there's some other random diffs still remaining as shown [here](https://github.com/cosmos/ibc-go/compare/main...fix/revert-02-client-changes#diff-88ffe4bc40902398b2e26d8d49ef88bf8c040f7b7cbcedd07918fc0d6d06af07R82) (scroll to `testing/` directory changes). I'm okay to revert any of these fixes later if needed, but also happy to get them in now

I'll need to figure the workaround to the testing pkg in #4804 for creating wasm clients

File: modules/light-clients/08-wasm/types/update_test.go (hunk 2 above)
Comment (charleenfei) [reply modules/light-clients/08-wasm/types/update_test.go:655]: probably just replicated the tm light client tests -- i can remove completely!

**Comment by charleenfei** [comment]
> Looks like [we can also clean up](https://github.com/cosmos/ibc-go/pull/4603/files#r1339866138) the [docker file](https://github.com/cosmos/ibc-go/blob/feat/wasm-clients/Dockerfile#L12-L16).

I will open an issue for this, I'd prefer to keep the changes in this PR to code cleanup/deletion

File: testing/wasm.go (hunk 3 above)
Comment (damiannolan) [reply testing/wasm.go:1]: +1, we can delete

**Review by damiannolan — APPROVED** [review]
lgtm

File: modules/light-clients/08-wasm/types/types_test.go [hunk 4]
```diff
@@ -21,41 +20,40 @@ import (
...
 	ctx      sdk.Context
 	store    sdk.KVStore
 	codeHash []byte
 	testData map[string]string
 }
 
-func init() {
-	ibctesting.DefaultTestingAppInit = setupTestingApp
-}
+// func init() {
```
Comment (DimitrisJim) [inline modules/light-clients/08-wasm/types/types_test.go:54]: QQ: was there a failure here? We need this `init` func to set up the correct simapp (one in 08-wasm)

Comment (charleenfei) [reply modules/light-clients/08-wasm/types/types_test.go:54]: you are right, i'll add it back in the next pr. i'm just confused why the tests didn't fail on this pr run 😕

**Comment by charleenfei** [comment]
> LGTM! I pushed a commit commenting out a bit of code that is also not used anymore.

this func was actually still being used (default `init()`) to create the simapp for the grandpa tests -- we probably need to think of a better way of changing this than relying on this default func

**Pull request merged**